src/main/java/com/segment/analytics/android/integrations/firebase/FirebaseIntegration.java

```diff
@@ -3,88 +3,278 @@
+      CharSequence activityLabel = info.loadLabel(packageManager);
+      /** setCurrentScreen should only be called in the onResume() activity */
+      firebaseAnalytics.setCurrentScreen(
+              activity, activityLabel.toString(), null /* class override */);
+      logger.verbose(
+              "firebaseAnalytics.setCurrentScreen(%s, %s, null /* class override */);",
+              activity, activityLabel.toString());
 
-  public FirebaseIntegration(Context context) {
-    mFirebaseAnalytics = FirebaseAnalytics.getInstance(context);
+    } catch (PackageManager.NameNotFoundException e) {
+      throw new AssertionError("Activity Not Found: " + e.toString());
+    }
   }
 
-  @Override public void onActivityStarted(Activity activity) {
-    super.onActivityStarted(activity);
+  @Override
+  public void identify(IdentifyPayload identify) {
+    super.identify(identify);
 
+    if (!isNullOrEmpty(identify.userId())) {
+      firebaseAnalytics.setUserId(identify.userId());
+      Map<String, Object> traits = identify.traits();
+      for (Map.Entry<String, Object> entry : traits.entrySet()) {
+        String trait = entry.getKey();
+        trait = trait.trim().replaceAll(" ", "_");
+        if (trait.length() > 40) {
+          trait = trimKey(trait);
+        }
+        String formattedValue;
+        if (entry.getValue() instanceof Date) {
+          Date value = (Date) entry.getValue();
+          formattedValue = formatDate(value);
+        } else if (entry.getValue() instanceof String && ISO_DATE.matcher(String.valueOf(entry.getValue())).matches()) {
+          formattedValue = String.valueOf(entry.getValue()).substring(0, 10);
+        } else {
+          formattedValue = String.valueOf(entry.getValue());
+        }
+
+        firebaseAnalytics.setUserProperty(trait, formattedValue);
+        logger.verbose("firebaseAnalytics.setUserProperty(%s, %s);", trait, formattedValue);
+      }
+    }
   }
 
-  @Override public void onActivityStopped(Activity activity) {
-    super.onActivityStopped(activity);
+  @Override
+  public void screen(ScreenPayload screen) {
+    super.screen(screen);
 
+    logger.verbose(
+            "The Firebase SDK gathers screen calls automatically. Segment does map manual screen " +
+                    "methods to Firebase.");
   }
 
-  @Override public void identify(IdentifyPayload identify) {
-    super.identify(identify);
+  @Override
+  public void track(TrackPayload track) {
+    super.track(track);
 
-    if (!isNullOrEmpty(identify.userId())) {
-      mFirebaseAnalytics.setUserId(identify.userId());
+    String event = track.event();
+    String eventName = mapEvent(event);
+    Properties properties = track.properties();
+
+    try {
+      Bundle formattedProperties = formatProperties(properties);
+      firebaseAnalytics.logEvent(eventName, formattedProperties);
+      logger.verbose("firebaseAnalytics.logEvent(%s, %s);", eventName, formattedProperties);
+    } catch (Exception e) {
```

Comment:
This should not be catching an exception as mentioned in https://github.com/segment-integrations/analytics-android-integration-firebase/pull/1#discussion_r137363288